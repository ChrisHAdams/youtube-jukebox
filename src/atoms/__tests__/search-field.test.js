import React from "react"
import renderer from "react-test-renderer"
import {render, fireEvent} from 'react-testing-library'

import SearchField from "../search-field";


describe("SearchField", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<SearchField/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

it('calls "onClick" prop on button click', () => {
  // Render new instance in every test to prevent leaking state
  let disabled = false;
  let searchValue = '';
  let searchValueChange = (event)=>{
    //console.log("New search value " + event.target.value);
    searchValue = event.target.value;
  }

  const { getByTestId } = render(<SearchField
                                  searchValueChange = {searchValueChange}
                                  searchValue = {searchValue}
                                  disable = {disabled} />);

  fireEvent.change(getByTestId('searchField'), { target: { value: 'TEST VALUE' } });
  expect(searchValue).toEqual('TEST VALUE');

  fireEvent.change(getByTestId('searchField'), { target: { value: 'New TEST VALUE' } });
  expect(searchValue).toEqual('New TEST VALUE');

});
