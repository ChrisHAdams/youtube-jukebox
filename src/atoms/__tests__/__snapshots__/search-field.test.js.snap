// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField renders correctly 1`] = `
<div>
  <input
    className="sc-bdVaJa fIgOID"
    data-testid="searchField"
    disabled={true}
    name="searchField"
    placeholder="Search for videos"
    type="text"
  />
</div>
`;
